x-pack/legacy/plugins/spaces/public/lib/hooks/use_spaces.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { kfetch } from 'ui/kfetch';
+import { Space } from '../../../common/model/space';
+import { GetSpacePurpose } from '../../../common/model/types';
+
+export const useKibanaSpaces = (purpose?: GetSpacePurpose) => {
+  const [spaces, setSpaces] = useState([] as Space[]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    kfetch({ pathname: '/api/spaces/space', query: { purpose } }).then((response: Space[]) => {
```

Comment:
Yeah ideally. Passing the `SpacesManager` into the hook would be problematic I think...it's awkward to pass it as a dependency of the hook, because then the hook really isn't doing anything important...it's just calling the `SpacesManager` for you.

I think I'll just remove this hook and replace it with a direct use of `useEffect` within the flyout to take advantage of the `SpacesManager`.